### DISCUSSION TITLE
Intent to remove: `overrides` folder

### DISCUSSION THREAD

**Discussion by RDIL** [thepeacockproject/Peacock#81] [Polls]
Hey all,
So for anybody who doesn't know, we currently have this thing in the configuration system where you can override specific built-in configs by creating a JSON file with the same name in the `overrides` folder.  It's been in Peacock since v1, and with the plugin system becoming a lot more mature, I think we should remove this feature as it has now outlived its usefulness.

**Comment by Burn-the-witch** [comment]
I think removing it is fine, as long as everything is exposed to plugins, so functionality isn't lost.

**Reply by RDIL** [reply]
@The-Jaded-Obsidian-Witch everything in the config system is already fully exposed via `controller.configManager`.

**Comment by LennardF1989** [comment]
Don't forget that due to the performance fixes we did, we might need to add some convenience function to add/replace additional config files. Testing it from my own plugin now.

**Reply by LennardF1989** [reply]
Yep, just tested. If you run Peacock, it will import static JSON files as strings, not parsed objects. So you need to do something like this to add/replace:

When running a plugin in Dev-mode, you need to do this:
```ts
import GenericMissionEndTemplate from "../static/GenericMissionEnd.json"

this.controller.configManager.configs["GenericMissionEndTemplate"] = GenericMissionEndTemplate
this.controller.configManager.configs["GenericMissionEndTemplate"] = JSON.parse(<string>this.controller.configManager.configs["GenericMissionEndTemplate"])
```

Can't quite get imports working as a .ts file for the build version tho, so not sure what happens there.
